pageserver/src/tenant/tasks.rs

```diff
@@ -173,9 +176,19 @@ async fn compaction_loop(tenant: Arc<Tenant>, cancel: CancellationToken) {
             } else {
                 // Run compaction
                 if let Err(e) = tenant.compaction_iteration(&cancel, &ctx).await {
```

Comment:
should e be matched to CompactionError or any error that correlates to walredo failure ?
if not this will backoff for download errors etc.

Reply:
Downloads have a local retry loop that should absorb those issues -- if they bubble up to this level, then we have some systemic problem with S3 & backing off is probably a good  thing.